x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/DataFrameTransformConfig.java

```diff
@@ -88,35 +92,50 @@ private static void validateStrictParsingParams(Object arg, String parameterName
                     SourceConfig source = (SourceConfig) args[1];
                     DestConfig dest = (DestConfig) args[2];
 
-                    SyncConfig syncConfig = (SyncConfig) args[3];
-                    // ignored, only for internal storage: String docType = (String) args[4];
+                    TimeValue frequency =
+                        args[3] == null ? null : TimeValue.parseTimeValue((String) args[3], DataFrameField.FREQUENCY.getPreferredName());
+                    if (lenient == false && frequency != null) {
+                        if (frequency.compareTo(MIN_FREQUENCY) < 0) {
+                            throw new IllegalArgumentException(
```

Comment:
I'm not a huge fan of performing this validation in the parser. The strict parser is used in PUT and preview `PutDataFrameTransformAction.Request` has a `validate` method as does `PreviewDataFrameTransformAction.Request` I think it's better to have all the validation in one place.

Reply:
Good idea.  I move the validation to the `validate` method of `PutDataFrameTransformAction.Request`.

`PreviewDataFrameTransformAction.Request`'s `validate` method was not validating nearly as much, so I left it out of there.  I guess that raises a wider question of whether preview should be validating parts of the config that aren't used in any way when previewing.